diff --git a/node_modules/react-native-smartrefreshlayout/SmartRefreshControl.js b/node_modules/react-native-smartrefreshlayout/SmartRefreshControl.js
index 9ed44ff..36869c7 100644
--- a/node_modules/react-native-smartrefreshlayout/SmartRefreshControl.js
+++ b/node_modules/react-native-smartrefreshlayout/SmartRefreshControl.js
@@ -14,7 +14,6 @@ import {ViewPropTypes} from './Util'
 import DefaultHeader from "./DefaultHeader";
 import PropTypes from 'prop-types';
 import processColor from 'react-native/Libraries/StyleSheet/processColor';
-import deprecatedPropType from 'react-native/Libraries/Utilities/deprecatedPropType'
 
 const SPModule =Platform.OS === 'android' ? NativeModules.SpinnerStyleModule : {};
 
@@ -147,7 +146,7 @@ SmartRefreshControl.propTypes = {
     onReleaseToRefresh:PropTypes.func,
     onHeaderReleased:PropTypes.func,
     enableRefresh: PropTypes.bool,//是否启用下拉刷新功能
-    HeaderComponent:deprecatedPropType(PropTypes.object,'Use the `renderHeader` prop instead.'),
+    HeaderComponent:PropTypes.object,
     renderHeader:PropTypes.oneOfType([
         PropTypes.func,
         PropTypes.element,
