diff --git a/node_modules/@sentry/react/cjs/errorboundary.js b/node_modules/@sentry/react/cjs/errorboundary.js
index 1b5d2e0..c6052f7 100644
--- a/node_modules/@sentry/react/cjs/errorboundary.js
+++ b/node_modules/@sentry/react/cjs/errorboundary.js
@@ -83,7 +83,7 @@ class ErrorBoundary extends React__namespace.Component {
   }
 
    componentDidCatch(error, { componentStack }) {
-    const { beforeCapture, onError, showDialog, dialogOptions } = this.props;
+    const { beforeCapture, onError, showDialog, dialogOptions, enableDefaultCaptureException = true } = this.props;
     browser.withScope(scope => {
       // If on React version >= 17, create stack trace from componentStack param and links
       // to to the original error using `error.cause` otherwise relies on error param for stacktrace.
@@ -106,14 +106,17 @@ class ErrorBoundary extends React__namespace.Component {
         beforeCapture(scope, error, componentStack);
       }
 
-      const eventId = browser.captureException(error, {
-        captureContext: {
-          contexts: { react: { componentStack } },
-        },
-        // If users provide a fallback component we can assume they are handling the error.
-        // Therefore, we set the mechanism depending on the presence of the fallback prop.
-        mechanism: { handled: !!this.props.fallback },
-      });
+      let eventId = null
+      if (enableDefaultCaptureException) {
+        eventId = browser.captureException(error, {
+          captureContext: {
+            contexts: { react: { componentStack } },
+          },
+          // If users provide a fallback component we can assume they are handling the error.
+          // Therefore, we set the mechanism depending on the presence of the fallback prop.
+          mechanism: { handled: !!this.props.fallback },
+        });
+      }
 
       if (onError) {
         onError(error, componentStack, eventId);
