diff --git a/node_modules/@sentry/react-native/android/src/main/java/io/sentry/react/RNSentryModuleImpl.java b/node_modules/@sentry/react-native/android/src/main/java/io/sentry/react/RNSentryModuleImpl.java
index 3c8a765..090c080 100644
--- a/node_modules/@sentry/react-native/android/src/main/java/io/sentry/react/RNSentryModuleImpl.java
+++ b/node_modules/@sentry/react-native/android/src/main/java/io/sentry/react/RNSentryModuleImpl.java
@@ -102,6 +102,7 @@ public class RNSentryModuleImpl {
     private final PackageInfo packageInfo;
     private FrameMetricsAggregator frameMetricsAggregator = null;
     private boolean androidXAvailable;
+    private Activity frameMetricsActivity = null;
 
     private static boolean hasFetchedAppStart;
 
@@ -402,6 +403,8 @@ public class RNSentryModuleImpl {
      * Returns frames metrics at the current point in time.
      */
     public void fetchNativeFrames(Promise promise) {
+        ensureFrameMetricsAggregatorAttached();
+
         if (!isFrameMetricsAggregatorAvailable()) {
             promise.resolve(null);
         } else {
@@ -678,34 +681,14 @@ public class RNSentryModuleImpl {
     }
 
     public void enableNativeFramesTracking() {
-        androidXAvailable = checkAndroidXAvailability();
-
-        if (androidXAvailable) {
-            frameMetricsAggregator = new FrameMetricsAggregator();
-            final Activity currentActivity = getCurrentActivity();
-
-            if (frameMetricsAggregator != null && currentActivity != null) {
-                try {
-                    frameMetricsAggregator.add(currentActivity);
-
-                    logger.log(SentryLevel.INFO, "FrameMetricsAggregator installed.");
-                } catch (Throwable ignored) {
-                    // throws ConcurrentModification when calling addOnFrameMetricsAvailableListener
-                    // this is a best effort since we can't reproduce it
-                    logger.log(SentryLevel.ERROR, "Error adding Activity to frameMetricsAggregator.");
-                }
-            } else {
-                logger.log(SentryLevel.INFO, "currentActivity isn't available.");
-            }
-        } else {
-            logger.log(SentryLevel.WARNING, "androidx.core' isn't available as a dependency.");
-        }
+        ensureFrameMetricsAggregatorAttached();
     }
 
     public void disableNativeFramesTracking() {
         if (isFrameMetricsAggregatorAvailable()) {
             frameMetricsAggregator.stop();
             frameMetricsAggregator = null;
+            frameMetricsActivity = null;
         }
     }
 
@@ -940,4 +923,42 @@ public class RNSentryModuleImpl {
     private boolean isFrameMetricsAggregatorAvailable() {
         return androidXAvailable && frameMetricsAggregator != null;
     }
+
+    private void ensureFrameMetricsAggregatorAttached() {
+        androidXAvailable = checkAndroidXAvailability();
+        if (!androidXAvailable) {
+            logger.log(SentryLevel.WARNING, "androidx.core' isn't available as a dependency.");
+            return;
+        }
+
+        if (frameMetricsAggregator == null) {
+            frameMetricsAggregator = new FrameMetricsAggregator();
+        }
+
+        final Activity currentActivity = getCurrentActivity();
+        if (currentActivity == null) {
+            logger.log(SentryLevel.INFO, "currentActivity isn't available.");
+            return;
+        }
+
+        if (frameMetricsActivity == currentActivity) {
+            return;
+        }
+
+        try {
+            if (frameMetricsActivity != null) {
+                frameMetricsAggregator.remove(frameMetricsActivity);
+                frameMetricsAggregator.reset();
+            }
+
+            frameMetricsAggregator.add(currentActivity);
+            frameMetricsActivity = currentActivity;
+
+            logger.log(SentryLevel.INFO, "FrameMetricsAggregator installed.");
+        } catch (Throwable ignored) {
+            // throws ConcurrentModification when calling addOnFrameMetricsAvailableListener
+            // this is a best effort since we can't reproduce it
+            logger.log(SentryLevel.ERROR, "Error adding Activity to frameMetricsAggregator.");
+        }
+    }
 }
diff --git a/node_modules/@sentry/react-native/dist/js/tracing/reactnativetracing.js b/node_modules/@sentry/react-native/dist/js/tracing/reactnativetracing.js
index 8c17268..27f9b4f 100644
--- a/node_modules/@sentry/react-native/dist/js/tracing/reactnativetracing.js
+++ b/node_modules/@sentry/react-native/dist/js/tracing/reactnativetracing.js
@@ -54,7 +54,7 @@ export class ReactNativeTracing {
             // eslint-disable-next-line deprecation/deprecation
             tracingOrigins, shouldCreateSpanForRequest, 
             // eslint-disable-next-line deprecation/deprecation
-            tracePropagationTargets: thisOptionsTracePropagationTargets, routingInstrumentation, enableAppStartTracking, enableStallTracking, } = this.options;
+            tracePropagationTargets: thisOptionsTracePropagationTargets, routingInstrumentation, enableAppStartTracking, enableStallTracking, lcpTrackingInstrumentation } = this.options;
             this._getCurrentHub = getCurrentHub;
             const clientOptionsTracePropagationTargets = clientOptions && clientOptions.tracePropagationTargets;
             // There are three ways to configure tracePropagationTargets:
@@ -82,7 +82,11 @@ export class ReactNativeTracing {
             }
             this._enableNativeFramesTracking(addGlobalEventProcessor, clientOptions);
             if ((_b = clientOptions === null || clientOptions === void 0 ? void 0 : clientOptions.enableStallTracking) !== null && _b !== void 0 ? _b : enableStallTracking) {
-                this.stallTrackingInstrumentation = new StallTrackingInstrumentation();
+                // this.stallTrackingInstrumentation = new StallTrackingInstrumentation();
+            }
+            if (lcpTrackingInstrumentation) {
+                logger.log('[ReactNativeTracing] LCP Tracking in React Native.');
+                this.lcpTrackingInstrumentation = lcpTrackingInstrumentation;
             }
             if (routingInstrumentation) {
                 routingInstrumentation.registerRoutingInstrumentation(this._onRouteWillChange.bind(this), this.options.beforeNavigate, this._onConfirmRoute.bind(this));
@@ -102,12 +106,13 @@ export class ReactNativeTracing {
     /**
      * To be called on a transaction start. Can have async methods
      */
-    onTransactionStart(transaction) {
-        var _a, _b;
+    onTransactionStart(transaction, payload) {
+        var _a, _b, _c;
         if (isNearToNow(transaction.startTimestamp)) {
             // Only if this method is called at or within margin of error to the start timestamp.
-            (_a = this.nativeFramesInstrumentation) === null || _a === void 0 ? void 0 : _a.onTransactionStart(transaction);
-            (_b = this.stallTrackingInstrumentation) === null || _b === void 0 ? void 0 : _b.onTransactionStart(transaction);
+            (_a = this.nativeFramesInstrumentation) === null || _a === void 0 ? void 0 : _a.onTransactionStart(transaction, payload);
+            (_b = this.stallTrackingInstrumentation) === null || _b === void 0 ? void 0 : _b.onTransactionStart(transaction, payload);
+            (_c = this.lcpTrackingInstrumentation) === null || _c === void 0 ? void 0 : _c.onTransactionStart(transaction, payload);
         }
         else {
             logger.warn(`[ReactNativeTracing] onTransactionStart called with delay (larger than margin of error) for transaction ${transaction.description} (${transaction.spanContext().spanId}). Not fetching native frames or tracking stalls.`);
@@ -117,9 +122,10 @@ export class ReactNativeTracing {
      * To be called on a transaction finish. Cannot have async methods.
      */
     onTransactionFinish(transaction, endTimestamp) {
-        var _a, _b;
+        var _a, _b, _c;
         (_a = this.nativeFramesInstrumentation) === null || _a === void 0 ? void 0 : _a.onTransactionFinish(transaction);
         (_b = this.stallTrackingInstrumentation) === null || _b === void 0 ? void 0 : _b.onTransactionFinish(transaction, endTimestamp);
+        (_c = this.lcpTrackingInstrumentation) === null || _c === void 0 ? void 0 : _c.onTransactionFinish(transaction, endTimestamp);
     }
     /**
      * Called by the ReactNativeProfiler component on first component mount.
@@ -389,8 +395,8 @@ export class ReactNativeTracing {
         }
     }
     /** To be called when the route changes, but BEFORE the components of the new route mount. */
-    _onRouteWillChange(context) {
-        return this._createRouteTransaction(context);
+    _onRouteWillChange(context, payload) {
+        return this._createRouteTransaction(context , payload);
     }
     /**
      * Creates a breadcrumb and sets the current route as a tag.
@@ -422,7 +428,7 @@ export class ReactNativeTracing {
         });
     }
     /** Create routing idle transaction. */
-    _createRouteTransaction(context) {
+    _createRouteTransaction(context, payload) {
         if (!this._getCurrentHub) {
             logger.warn(`[ReactNativeTracing] Did not create ${context.op} transaction because _getCurrentHub is invalid.`);
             return undefined;
@@ -435,7 +441,7 @@ export class ReactNativeTracing {
         const { finalTimeoutMs } = this.options;
         const expandedContext = Object.assign(Object.assign({}, context), { trimEnd: true });
         const idleTransaction = this._startIdleTransaction(expandedContext);
-        this.onTransactionStart(idleTransaction);
+        this.onTransactionStart(idleTransaction, payload);
         logger.log(`[ReactNativeTracing] Starting ${context.op} transaction "${context.name}" on scope`);
         idleTransaction.registerBeforeFinishCallback((transaction, endTimestamp) => {
             this.onTransactionFinish(transaction, endTimestamp);
@@ -455,7 +461,7 @@ export class ReactNativeTracing {
                 var _a, _b;
                 if (
                 // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-                ((_b = (_a = transaction.data) === null || _a === void 0 ? void 0 : _a.route) === null || _b === void 0 ? void 0 : _b.hasBeenSeen) &&
+                ((_b = (_a = transaction.data) === null || _a === void 0 ? void 0 : _a.route) === null || _b === void 0 ? void 0 : _b.hasBeenSeen) ||
                     (!transaction.spanRecorder ||
                         transaction.spanRecorder.spans.filter(span => span.spanId !== transaction.spanId &&
                             span.op !== 'ui.load.initial_display' &&
